src/adaptors/tarot/index.js

```diff
@@ -0,0 +1,650 @@
+        lendingPoolDecimals,
+        lendingPoolReserves,
+      } = await getAllLendingPools(factory, chain, block);
+      // get underlying LP related info
+      const {
+        lendingPoolSymbols,
+        underlyingLpAddresses,
+        underlyingLiquidityPoolSymbols,
+      } = await getUnderlyingLiquidityPoolAddresses(
+        lendingPoolAddresses,
+        lendingPoolAddressesTargetCalls,
+        chain,
+        block
+      );
+      const allTokens = [...token0s, ...token1s];
+      const allUniqueTokens = removeDuplicatesFromArray(allTokens);
+      const coinKeys = allUniqueTokens.map((token) => {
+        let transformedToken = transformTokenForApi(chain, token, transform);
+        return transformedToken;
+      });
+      let fetchedPrices = await getPrices(coinKeys);
+      // go to the next factory contract since no tokens were found here.
+      if (Object.keys(fetchedPrices) === 0) {
+        continue;
+      }
+      const tokenDetailsDict = await getTokenDetails(
+        allUniqueTokens,
+        chain,
+        block
+      );
+      // loop through lending pool
+      for (let i = 0; i < lendingPoolsDetails.length; i++) {
+        let lendingPoolAddress = lendingPoolAddresses[i];
+        // check if pool is disabled
+        if (
+          DISABLED_LENDING_POOLS.indexOf(lendingPoolAddress.toLowerCase()) !==
+          -1
+        ) {
+          continue;
+        }
+        let lendingPool = lendingPoolsDetails[i];
+        const underlyingLpAddress = underlyingLpAddresses[i];
+        // get underlying (LP) of lending pool symbol
+        const underlyingLiquidityPoolSymbol = underlyingLiquidityPoolSymbols[i];
+        // check if the lending pool has a name
+        if (underlyingLiquidityPoolSymbol === null) {
+          continue;
+        }
+        let borrowables = [];
+        borrowables.push(lendingPool.borrowable0);
+        borrowables.push(lendingPool.borrowable1);
+        let collateral = lendingPool.collateral;
+        let tokens = [];
+        tokens = [token0s[i], token1s[i]];
+        const lendingPoolDecimal = lendingPoolDecimals[i];
+        const reserves = lendingPoolReserves[i];
+        // ignore lending pools with no reserves because you can not calculate the APY and TVL.
+        if (reserves == null) {
+          continue;
+        }
+        // loop through each borrowable tokens and push apy/tvl data
+        for (let j = 0; j < borrowables.length; j++) {
+          borrowableTokenAddress = borrowables[j];
+          underlyingTokenAddress = tokens[j];
+          tokenSymbol = tokenDetailsDict[underlyingTokenAddress].symbol;
+          tokenDecimals = tokenDetailsDict[underlyingTokenAddress].decimals;
+          lpReserve = reserves[j];
+          const {
+            excessSupply,
+            reserveFactor,
+            totalBorrows,
+            borrowRate,
+            borrowableDecimal,
+            totalSupply,
+          } = await getUnderlyingTokenAndBorrowableDetails(
```

Comment:
u can make requests in this for loop parallel 
for example `borrowables.map( (smth) => getUnderlyingTokenAndBorrowableDetails().then((res => transformTarotLPName()) )` , so that way they won't wait for each other's execution

Reply:
@vrtnd Thanks for your feedback! Yea you're right it's not that optimized... I'll try to do a map function as you suggested and get the lending pool related data before I start iterating them to calculate the apy/tvl. Sorry my branch was stale too so I updated it earlier and I'll add the fields that you suggested above.